torch/testing/__init__.py

```diff
@@ -6,9 +6,56 @@
 import random
 
 __all__ = [
-    'make_non_contiguous', 'rand_like', 'randn_like'
+    'assert_allclose', 'make_non_contiguous', 'rand_like', 'randn_like'
 ]
 
+rand_like = torch._C._VariableFunctions.rand_like
+randn_like = torch._C._VariableFunctions.randn_like
+
+
+def assert_allclose(actual, expected, rtol=None, atol=None, equal_nan=True):
+    if not isinstance(actual, torch.Tensor):
+        actual = torch.tensor(actual)
+    if not isinstance(expected, torch.Tensor):
+        expected = torch.tensor(expected, dtype=actual.dtype)
+    if expected.shape != actual.shape:
+        expected = expected.expand_as(actual)
+    if rtol is None or atol is None:
+        if rtol is not None or atol is not None:
+            raise ValueError("rtol and atol must both be specified or both be unspecified")
+        rtol, atol = _get_default_tolerance(actual, expected)
+
+    close = torch.isclose(actual, expected, rtol, atol, equal_nan)
+    if close.all():
+        return
+
+    # Find the worst offender
+    error = (expected - actual).abs()
+    expected_error = atol + rtol * expected.abs()
+    delta = error - expected_error
+    delta[close] = 0  # mask out NaN/inf
+    _, index = delta.reshape(-1).max(0)
+
+    # TODO: consider adding torch.unravel_index
+    def _unravel_index(index, shape):
+        res = []
+        for size in shape[::-1]:
+            res.append(int(index % size))
+            index = int(index // size)
+        return tuple(res[::-1])
+
+    index = _unravel_index(index.item(), actual.shape)
+
+    # Count nubmer of offenders
+    count = (~close).long().sum()
+
+    msg = ('Not within tolerance rtol={} atol={} at input{} ({} vs. {}) and {}'
```

Comment:
at input -> at index?

Reply:
It prints something like:

```
AssertionError: Not within tolerance rtol=0.0001 atol=1e-05 at input[1, 0] (-1.6947613954544067 vs. 1.5197489261627197) and 11 other locations (100.00%)
```

So it looks like your indexing a tensor named `input`